runtime.go

```diff
@@ -585,9 +587,18 @@ func (r *Runtime) MemUsage(ctx *MemUsageContext) (memUsage uint64, err error) {
 	return memUsage, nil
 }
 
-// MaxStackObjectMemUsage returns the memory used by the largest object seen in the vm value stack
-func (r *Runtime) MaxStackObjectMemUsage() uint64 {
-	return r.maxStackObjectMem
+// GetMaxStackObjectMemUsage returns the memory used by the largest object seen in the vm value stack
+func (r *Runtime) GetMaxStackObjectMemUsage() uint64 {
+	r.maxStackObjectMemLock.RLock()
+	mem := r.maxStackObjectMem
+	r.maxStackObjectMemLock.RUnlock()
+	return mem
+}
+
+func (r *Runtime) setMaxStackObjectMemUsage(mem uint64) {
+	r.maxStackObjectMemLock.Lock()
+	r.maxStackObjectMem = mem
```

Comment:
Logic here is a bit different, not sure it makes a difference (e.g. not using math.Max)

Reply:
yep, just using this as a setter here! And used a logical check over Math.max elsewhere to reduce the number of times we call into this  to get the lock